src/query/plan/operator.cpp

```diff
@@ -237,6 +239,10 @@ CreateNode::CreateNodeCursor::CreateNodeCursor(const CreateNode &self, utils::Me
 bool CreateNode::CreateNodeCursor::Pull(Frame &frame, ExecutionContext &context) {
   SCOPED_PROFILE_OP("CreateNode");
 
+  if (context.auth_checker && !context.auth_checker->Accept(*context.db_accessor, self_.node_info_.labels,
+                                                            memgraph::auth::FineGrainedPermission::CREATE_DELETE))
+    return false;
```

Comment:
Check with Boris if we need to rather throw an exception here.

Reply:
I think Buda mentioned about some feedback message that some of the vertices were not removed, like an info message that comes with the query

Reply:
I think this not the best approach as running a create query without the correct permissions will run without any errors but the users won't see any result. Compared to other privileges, the query will fail with the error that hey, you don't have permission to do that. However this is only my opinion, feel free to discard it.